c/tskit/tables.c

```diff
@@ -4715,6 +4716,152 @@ tsk_table_sorter_free(tsk_table_sorter_t *self)
     return 0;
 }
 
+/******************
+ * Canonical sort
+ ******************/
+
+typedef struct {
+    tsk_mutation_t mut;
+    int num_descendants;
+} mutation_canonical_sort_t;
+
+static int
+cmp_mutation_canonical(const void *a, const void *b)
+{
+    const mutation_canonical_sort_t *ia = (const mutation_canonical_sort_t *) a;
+    const mutation_canonical_sort_t *ib = (const mutation_canonical_sort_t *) b;
+    /* Compare mutations by site */
+    int ret = (ia->mut.site > ib->mut.site) - (ia->mut.site < ib->mut.site);
+    if (ret == 0 && !tsk_is_unknown_time(ia->mut.time)
+        && !tsk_is_unknown_time(ib->mut.time)) {
+        ret = (ia->mut.time < ib->mut.time) - (ia->mut.time > ib->mut.time);
+    }
+    if (ret == 0) {
+        ret = (ia->num_descendants < ib->num_descendants)
+              - (ia->num_descendants > ib->num_descendants);
+    }
+    if (ret == 0) {
+        ret = (ia->mut.node > ib->mut.node) - (ia->mut.node < ib->mut.node);
+    }
+    if (ret == 0) {
+        ret = (ia->mut.id > ib->mut.id) - (ia->mut.id < ib->mut.id);
```

Comment:
I don't know why Codecov says this is not covered - I've stuck in some printfs, and it IS covered, in `tests/test_tables.py::TestSortTables::test_no_mutation_parents`.

Reply:
Needs to be covered by C tests, Python tests aren't counted for coverage here.

Reply:
I added an identical mutation to your test_canonicalise. Nice test!